pkg/didcomm/transport/http/inbound.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package http
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+// InboundDIDCommHandler struct to pass details for request handlers
+type InboundDIDCommHandler struct {
+	Path        string
+	HandlerFunc http.HandlerFunc
+}
+
+// NewInboundHandler will create a new handler to enforce Did-Comm HTTP transport specs
+// then routes processing for the passed 'inURL' and 'handlerFunc' arguments.
+// Both 'inURL' and 'handlerFunc' are mandatory for inbound communications.
+func NewInboundHandler(inURL string, handlerFunc http.HandlerFunc) (http.Handler, error) {
+	commHandler := &InboundDIDCommHandler{Path: inURL, HandlerFunc: handlerFunc}
+	err := validateHandler(commHandler)
```

Comment:
it seems we are creating a struct just so it can be passed here? are you planning more extensive usage of this type?

Reply:
for now there's no plan to extend this type, it's been unexported